### PR TITLE
TextBox - reset Text when pushing null to bound value

### LINKED ISSUE #1302: TextBox does not clear when binding set to null
## Current behavior

After the recent changes in `TextBox` controls, it appears that when the data bound source is set to `null` the `TextBox` does not clear itself. Instead, it stays filled out with the previous input.

Example C#:

    public class TestItem
    {
        public string Text { get; set; } 
    }

    public TestItem Test { get; private set; } = new TestItem();

    private void ButtonBase_OnClick(object sender, RoutedEventArgs e)
    {
         Test = new TestItem();
         OnPropertyChanged(nameof(Test));
    }

and XAML:

    <StackPanel Padding="20" Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
        <TextBox Text="{x:Bind Test.Text, Mode=TwoWay}" />
        <Button Click="ButtonBase_OnClick" Content="Clear" />
    </StackPanel>
    

## Expected behavior

Same as UWP, the `TextBox` should clear when `Text` binding source is set to `null`

## How to reproduce it (as minimally and precisely as possible)

Simple repro app is here: https://github.com/MartinZikmund/UnoNullTextBoxBug

## Environment

Nuget Package:

Package Version(s):

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [x] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by davidjohnoliver**
Fix regression where a null pushed to Text via binding was being ignored, rather than clearing the Text.

GitHub Issue (If applicable): fixes #1302 

## PR Type

What kind of change does this PR introduce?

 - Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBoxTests/Given_TextBox.cs [hunk 1]
```diff
@@ -40,17 +44,45 @@ public void When_Setting_Text_Null()
...
+
 			Assert.AreEqual("Rhubarb", textBox.Text);
 			Assert.AreEqual(1, callbackCount);
 		}
 
+		[TestMethod]
+		public void When_Binding_Set_Null()
+		{
+			var textBox = new TextBox();
+			var source = new MySource() { SourceText = "Spinach" };
```
Comment (carldebilly) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBoxTests/Given_TextBox.cs]: Love your samples ;-)

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 2]
```diff
@@ -153,7 +153,14 @@ private DependencyPropertyChangedEventArgs CreateInitialValueChangerEventArgs(De
 		public string Text
 		{
 			get { return (string)this.GetValue(TextProperty); }
-			set { this.SetValue(TextProperty, value); }
+			set {
+				if (value == null)
+				{
+					throw new ArgumentNullException();
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs]: Is it the behavior of UWP? Surprising...

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs]: yep

**Pull request merged**